content/en/docs/how-tos/use-registries-in-build-farm.md

```diff
@@ -0,0 +1,118 @@
+  name: image-puller
+```
+
+After the pull request is merged, the yaml files will be applied automatically to `app.ci`. We can then get the token of, e.g., `SA/image-puller` in `namespace/ci`:
+
+```bash
+$ namespace=ci
+$ sa=image-puller
+$ hostname=image-registry.openshift-image-registry.svc:5000
+$ oc get secrets --namespace $namespace -o json | jq --arg sa $sa '.items[] | select(.type=="kubernetes.io/dockercfg") | select(.metadata.annotations["kubernetes.io/service-account.name"]==$sa) | .data[".dockercfg"]' --raw-output | base64 --decode | jq --arg h $hostname 'with_entries(select(.key==$h))'
+{
+  "image-registry.openshift-image-registry.svc:5000": {
+    "auth": "token"
+  }
+}
+```
+
+The token can be used to modify the `auth.json` file above.
+
+# Access the images in CI
+
+Determine which cluster the job is running on, e.g. [the follow line](https://prow.ci.openshift.org/view/gs/origin-ci-test/logs/branch-ci-opendatahub-io-odh-manifests-master-images/1329789689988648960#1:build-log.txt%3A8) in the log of job `branch-ci-opendatahub-io-odh-manifests-master-images` says that the job runs on `build02` and that `ci-op-2c2tvgti` is the temporary namespace for the test:
+
+```
+2020/11/20 14:12:28 Using namespace https://console.build02.ci.openshift.org/k8s/cluster/projects/ci-op-2c2tvgti
+```
+
+All the images used in test are stored in the imagestreams in `ci-op-2c2tvgti` which is also available as an artifact of the job: [imagestreams.json](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/logs/branch-ci-opendatahub-io-odh-manifests-master-images/1329789689988648960/artifacts/build-resources/imagestreams.json).
+
+The image for `istag/pipeline:odh-manifests-tests` in `namespace/ci-op-2c2tvgti` can be pulled by:
+
+```bash
+$ podman pull --authfile auth.json  registry.build02.ci.openshift.org/ci-op-2c2tvgti/pipeline:odh-manifests-tests
+```
+
+Note that the `auth.json` file has to contain a valid token for `registry.build02.ci.openshift.org` from a user or a service account which has permissions to pull the image.
+
+# Access the promoted images
+
+If `ci-operator`'s config has `promotion` stanza, e.g.,
+
+```yaml
+images:
+- dockerfile_path: images/hello-openshift/Dockerfile.rhel
+  from: base
+  inputs:
+    ocp_builder_rhel-8-golang-1.15-openshift-4.7:
+      as:
+      - registry.svc.ci.openshift.org/ocp/builder:rhel-8-golang-1.15-openshift-4.7
+  to: hello-openshift
+promotion:
+  name: "4.7"
+  namespace: ocp
+```
+
+The post-submit job, usually named with `branch-ci-<org>-<repo>-<branch>-images` pushes the images to the central CI registry at `app.ci`. We can pull the image defined by above `promotion` stanza by:
+
+```bash
+$ podman pull --authfile auth.json  registry.ci.openshift.org/ocp/4.7:hello-openshift
+```
+
+The promoted images are distributed to other clusters too, e.g., the same image can be pulled with `<any_registry_dns_in_build_farm>/ocp/4.7:hello-openshift`.
+
+# Hit an unauthenticated error
```

Comment:
Please add a `Troubleshooting` section instead where you describe what error is seen when and how to mitigate (here: I am getting an "unauthenticated" error, why? Response: You most likely logged in at some point and your token expired, this now prevents all pulls from that registry, even for public images

Reply:
Updated.